### PR TITLE
adding flashcards

### PR DESCRIPTION
**Pull request by agnieszkapawska**
Adding new flashcards.

### CODE REVIEW

File: src/main/java/com/agnieszkapawska/flashcards/domain/dtos/FlashcardDto.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class FlashcardDto {
+    private String question;
+    private String answer;
+    private String exampleUsage;
+    private String explanation;
+    private Set<QuestionTags> questionTagsList;
```
Comment (krzysztofpawski) [inline src/main/java/com/agnieszkapawska/flashcards/domain/dtos/FlashcardDto.java]:
```suggestion
    private Set<QuestionTags> tagsList;
```
I think it's more like flashcardTagsList but IMO tagsList is enough. We are in this class and we know that those are properties of this class. We don't need to write it in property name.

File: src/main/java/com/agnieszkapawska/flashcards/domain/dtos/FlashcardResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.agnieszkapawska.flashcards.domain.dtos;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class FlashcardResponseDto {
```
Comment (krzysztofpawski) [inline src/main/java/com/agnieszkapawska/flashcards/domain/dtos/FlashcardResponseDto.java]:
```suggestion
public class FlashcardSaveResponseDto {
```
I guess there will be also response for updating and deleting. Do you want to use the same response or you will have separate classes for all cases? I'm not sure what is better option, just asking :)

File: src/main/java/com/agnieszkapawska/flashcards/api/controllers/FlashcardController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/flashcard")
+public class FlashcardController {
+    private FlashcardFacade flashcardFacade;
+
+    @PostMapping(consumes = "application/json")
+    public ResponseEntity<FlashcardResponseDto> saveFlashcard(@RequestBody FlashcardDto flashcardDto) {
+        return new ResponseEntity<>(flashcardFacade.saveFlashcard(flashcardDto), HttpStatus.OK);
```
Comment (krzysztofpawski) [inline src/main/java/com/agnieszkapawska/flashcards/api/controllers/FlashcardController.java:22]: Is there any possibility that saving flascard will fail? Do we need some error handling here? I just don't remeber how did you implement in previous projects that for example in some cases you were returning 404 or something like this.

**Review by krzysztofpawski — DISMISSED** [review]
Just minor comments. Generally good job :)

File: src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.dao.DataIntegrityViolationException;
+
+@AllArgsConstructor
+public class FlashcardFacade {
+    private FlashcardRepository flashcardRepository;
+    private ModelMapper modelMapper;
+
+    public FlashcardSaveResponseDto saveFlashcard(FlashcardDto flashcardDto) {
+        Flashcard flashcard = modelMapper.map(flashcardDto, Flashcard.class);
+        try{
```
Comment (krzysztofpawski) [inline src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java:19]:
```suggestion
        try {
```
missing space

**Review by krzysztofpawski — APPROVED** [review]

File: src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java (hunk 4 above)
Comment (agnieszkapawska) [reply src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java:19]: I will do it in next PR

**Pull request merged**